### PR TITLE
Remove native_functions.yaml dependency from CUDA distributions

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #73162
* __->__ #67875

Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

### CODE REVIEW

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

**Review by dagitses — APPROVED** [review]

File: aten/src/ATen/native/cuda/Distributions.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#include <ATen/native/cuda/Distributions.h>
+#include <ATen/Functions.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/TensorIterator.h>
+#include <ATen/CUDAGeneratorImpl.h>
+
+namespace at { namespace native {
+
+Tensor _s_poisson_cuda(const Tensor& lambda, c10::optional<Generator> gen_) {
+  auto gen = get_generator_or_default<CUDAGeneratorImpl>(gen_, cuda::detail::getDefaultCUDAGenerator());
```
Comment (dagitses) [inline aten/src/ATen/native/cuda/Distributions.cpp:10]: these calls are flagging:
> Call to at::cuda::detail::getDefaultCUDAGenerator is logically private, please avoid use outside of its parent namespace

Do you think this is bogus or is there any refactoring that makes sense here?

Comment (peterbell10) [reply aten/src/ATen/native/cuda/Distributions.cpp:10]: It sounds like it doesn't like that a `detail` namespace is being used outside of the `at::cuda` namespace but that's just how ATen is structured.

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

**Comment by peterbell10** [comment]
@malfet ping

**Comment by malfet** [comment]
@peterbell10 this change break a bunch of Windows builds are `struct` and `class` are not interchangeable by VC++, let me try to rebase your change and see if it already fixed the problem, if not I'll suggest a small fix

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

**Comment by peterbell10** [comment]
I think I got it. Weird that open source CI didn't catch it though.

**Comment by malfet** [comment]
> I think I got it. Weird that open source CI didn't catch it though.

Yes, I believe this is it, giving PR another try

**Pull request closed without merging**